core/modules/entity/entity.controller.inc

```diff
@@ -291,6 +291,11 @@ class DefaultEntityController implements EntityControllerInterface {
         }
       }
       $query->fields('revision', $entity_revision_fields);
+
+      // Compare revision id of the base and revision table, if equal then this
+      // is the active revision.
+      $query->addExpression('base.' . $this->revisionKey . ' = revision.' . $this->revisionKey, 'is_active_revision');
+      $query->addField('base', $this->revisionKey, 'active_revision');
```

Comment:
It's possibly confusing that `$entity->active_revision` is read-only. Setting it doesn't have any result on the save operation. Is this property necessary?

Reply:
This is meant to set `active_revision` as an alias of `revisionKey` during loading the entity.